### DISCUSSION TITLE
Allow ??= operator on auto-properties

### DISCUSSION THREAD

**Discussion by Pentadome** [dotnet/csharplang#3458] [Language Ideas]
Allow ??= operator on auto-properties

Summary
The ??= currently is a shorthand for:
``` C#
_something ?? (_something = new Something(this));
// is the same as:
_something ??= new Something(this);
// However this operator can not be used on properties.

List<Something> Somethings { get; } ??= new List<Something>(); // Error

// Therefore we need to write this;

private List<Something> _somethings;

public List<Something> Somethings => _somethings ??= new List<Something>();

```

Motivation
Currently "lazy" properties take up too much space and require too much typing for something that should be able to be expressed more easily.

Detailed design
Allow ??= on properties. So that these two mean the same thing:

``` C#
public ICommand DoSomethingCommand { get; } ??= new Command(this.DoSomething);
private void DoSomething() {}
```
``` C#
private Command _doSomethingCommand;
public ICommand DoSomethingCommand => _doSomethingCommand ??= new Command(this.DoSomething);
private void DoSomething() {}
```

Drawbacks
People who don't know what `??=` does might be confused.

Unresolved questions
Should the backing field generated be accessible by other members of the class? If so, how? Like `DoSomethingCommand.field`, `fieldof(DoSomethingCommand)` or not at all?

**Comment by Joe4evr** [comment]
This is superseded by #140, which will allow you to write
```cs
public ICommand DoSomethingCommand => field ??= new Command(this.DoSomething);
```

**Comment by yaakov-h** [comment]
Seems like another use-case for #140 

```csharp
public List<Thing> Things => field ??= new List<Thing>();
```

**Comment by Pentadome** [comment]
@Joe4evr It is not approved, is it?

**Comment by Joe4evr** [comment]
It's championed, [discussed in LDM last month](https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-01.md#field-keyword-in-properties), and I think it was @jnm2 who did a lot of work for the feature already.

**Comment by jnm2** [comment]
Another one to add to https://gist.github.com/jnm2/d3b37a43894f1738273005e69824d089#140-mentioned-as-preferred-or-alleviating-the-pain

@Joe4evr I just started the prototype a few days ago and it looks like it will go relatively quickly
